rclcpp_lifecycle/src/lifecycle_node.cpp

```diff
@@ -155,15 +155,23 @@ LifecycleNode::~LifecycleNode()
   auto current_state = LifecycleNode::get_current_state().id();
   // shutdown if necessary to avoid leaving the device in any other primary state
   if (current_state < lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED) {
-    auto ret = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::ERROR;
-    auto finalized = LifecycleNode::shutdown(ret);
-    if (finalized.id() != lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED ||
-      ret != rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS)
-    {
-      RCLCPP_WARN(
+    if (node_base_->get_context()->is_valid()) {
+      auto ret = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::ERROR;
+      auto finalized = LifecycleNode::shutdown(ret);
+      if (finalized.id() != lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED ||
+        ret != rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS)
+      {
+        RCLCPP_WARN(
+          rclcpp::get_logger("rclcpp_lifecycle"),
+          "Shutdown error in destruction of LifecycleNode: final state(%s)",
+          finalized.label().c_str());
+      }
+    } else {
+      // TODO(fujitatomoya): consider when context is gracefully shutdown before.
+      RCLCPP_DEBUG(
```

Comment:
I would like to keep this `DEBUG` until we figure out the proper way to shutdown with invalid context.